### ISSUE TITLE
JWT Tokens unable to validate - Invalid KID

### ISSUE THREAD

**Issue by williamdenton** [auth0/auth0.net#293]
### Description

`Auth0.AuthenticationApi` references `Microsoft.IdentityModel.Protocols.OpenIdConnect v5.4.0` that shipped with a change to the way `kid` is calculated. 

Adding the nuget `Auth0.AuthenticationApi v6.5.2` to any existing service running `AspNet.Security.OpenIdConnect.Server v2.0.0` will perform an implicit upgrade on `Microsoft.IdentityModel.Protocols.OpenIdConnect` which changes the way the kid is generated. This will break clients issued with tokens from the service that use Auth0s supplied javascript library. 
>*note*: this does not apply to tokens issued by auth0

### Prerequisites

- [x] I have checked the [documentation for this library](https://auth0.github.io/auth0.net/).

- [x] I have checked the [Auth0 Community](https://community.auth0.com/) for related posts.

- [x] I have checked for related or duplicate [Issues](https://github.com/auth0/auth0.net/issues) and [PRs](https://github.com/auth0/auth0.net/pulls).

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Environment

Please provide the following:

- Version of Auth0.NET used: `"Auth0.AuthenticationApi" Version="6.5.1"`
- Version of .NET used: `dotnet core 2.1`
- Any other relevant environmental versions:
- Additional libraries that might be affecting your instance:
  * `AspNet.Security.OpenIdConnect.Server` v2.0.0
  * https://www.npmjs.com/package/auth0-js

### Reproduction

`Microsoft.IdentityModel.Protocols.OpenIdConnect` introduced a bug in v`5.4.0` that changed the value for the `kid` (Key ID). Previously the key was set to the thumb print of the certificate. This was fixe in v`5.5.0`

This is the commit that breaks the `keyid` (released in 5.4) https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/760fa3046cc36591c9f7835031745fcd625cf6ec#diff-ad9f507af4d8334a3a5dbcafbb0ded30R53

This commit restores the `keyid` to the thumbprint (released in v5.5) https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1140/files#diff-ad9f507af4d8334a3a5dbcafbb0ded30R53

`Auth0.AuthenticationApi` references the faulty 5.4 release
https://github.com/auth0/auth0.net/blob/f676441b5915923221e963b08715eae207cc3beb/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj#L20-L23

The fix for this issue is to update `Microsoft.IdentityModel.Protocols.OpenIdConnect` to version >= `5.5.0`  or downgraded to `5.3.0`. BUT READ ON, UPDATING WILL BREAK CLIENTS

If running a token server (ie not using auth0, or using auth0 _and_ your own STS server using `AspNet.Security.OpenIdConnect.Server`) taking a reference on this project will implicitly upgrade the reference to the broken version. This will break all existing javascript clients if they use the Auth0 supplied jwt validation code. This is because it uses an unsafe way to obtain the jwks  documented this  _long_ standing issue https://github.com/auth0/node-jwks-rsa/issues/5

In summary, to avoid breaking clients, Auth0 must fix the javascript npm package, then fix the package reference for this project.

For search-ability this is  the error messages seen from clients
>`Unable to find a signing key that matches`

You may think it strange referencing Auth0 from an application that is a Secure Token Server already, but we are in the process of migrating to Auth0 and this is the very real scenario we found ourselves in.

**Comment by williamdenton** [comment]
this is the issue in `Microsoft.IdentityModel.Protocols.OpenIdConnect ` where the `kid` issue was resolved https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1108

**Comment by damieng** [comment]
Unfortunately trying to upgrade to 5.5 of that library gives conflict
erroea with .net standard library. :(

**Comment by williamdenton** [comment]
Hi @damieng (thanks for the super fast response)

Just pulled your repo - strange, i see errors to when upgrading to 5.5/
```
/Users/williamdenton/Documents/GitHub/auth0.net/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj : error NU1605: Detected package downgrade: NETStandard.Library from 2.0.3 to 1.6.1. Reference the package directly from the project to select a different version.  [/Users/williamdenton/Documents/GitHub/auth0.net/Auth0.Net.sln]
/Users/williamdenton/Documents/GitHub/auth0.net/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj : error NU1605:  Auth0.AuthenticationApi -> Microsoft.IdentityModel.Protocols.OpenIdConnect 5.5.0 -> NETStandard.Library (>= 2.0.3)  [/Users/williamdenton/Documents/GitHub/auth0.net/Auth0.Net.sln]
/Users/williamdenton/Documents/GitHub/auth0.net/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj : error NU1605:  Auth0.AuthenticationApi -> NETStandard.Library (>= 1.6.1) [/Users/williamdenton/Documents/GitHub/auth0.net/Auth0.Net.sln]
/Users/williamdenton/Documents/GitHub/auth0.net/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj : error NU1605: Detected package downgrade: NETStandard.Library from 2.0.3 to 1.6.1. Reference the package directly from the project to select a different version.  [/Users/williamdenton/Documents/GitHub/auth0.net/Auth0.Net.sln]
/Users/williamdenton/Documents/GitHub/auth0.net/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj : error NU1605:  Auth0.AuthenticationApi -> System.IdentityModel.Tokens.Jwt 5.5.0 -> NETStandard.Library (>= 2.0.3)  [/Users/williamdenton/Documents/GitHub/auth0.net/Auth0.Net.sln]
/Users/williamdenton/Documents/GitHub/auth0.net/src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj : error NU1605:  Auth0.AuthenticationApi -> NETStandard.Library (>= 1.6.1) [/Users/williamdenton/Documents/GitHub/auth0.net/Auth0.Net.sln]
    0 Warning(s)
    2 Error(s)
```

however downgrading to `5.3` works.

However critically here is that the javascript library gets updated. If that was updated this wouldn't be such a big problem

**Comment by damieng** [comment]
Should be fixed in the upcoming release (tentantively Monday).
